app/controllers/menus_controller.rb

```diff
@@ -2,6 +2,11 @@ class MenusController < ApplicationController
 
   def show
     @restaurant = Restaurant.find(params[:restaurant_id])
-    @cart_nr = ShoppingCartItem.count
+    if session[:cart_id]
+      @cart = ShoppingCart.find(session[:cart_id])
+      @cart_nr = @cart.shopping_cart_items.count
```

Comment:
Why do you have `@cart_nr`? what is that used for? I initially thought it was used on a view somewhere but i can't see it anywhere on this PRs diff..

Even if you need the cart items counts, you could very well just use `@cart.shopping_cart_items.count`